src/elements/form/Control.ts

```diff
@@ -35,13 +35,13 @@ import { Helpers } from './../../utils/Helpers';
                     <!--Text-based Inputs-->
                     <!--TODO prefix/suffix on the control-->
                     <div class="novo-control-input-container novo-control-input-with-label" *ngSwitchCase="'textbox'">
-                        <input *ngIf="control.type !== 'number'" [formControlName]="control.key" [id]="control.key" [type]="control.type" [placeholder]="control.placeholder" (input)="emitChange($event)" [maxlength]="control.maxlength" (focus)="handleFocus($event)" (blur)="handleBlur($event)">
+                        <input *ngIf="control.type !== 'number'" [formControlName]="control.key" [id]="control.key" [type]="control.type" [placeholder]="control.placeholder" (input)="emitChange($event)" (keypress)="checkMaxLength($event)" [maxlength]="control.maxlength" (focus)="handleFocus($event)" (blur)="handleBlur($event)">
```

Comment:
Don't need keypress, you have the call inside emitChange, so you are calling it twice. Can remove the keypress event listenter.